lib/board.rb

```diff
@@ -0,0 +1,53 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(spot)
+    cells[spot.to_i - 1]
+  end
+
+  def full?
+    !(cells.include?(" "))
+  end
+
+  def turn_count
+    turns = cells.delete_if{|a| a == " "}
+    turns.count
+  end
+
+  def taken?(spot)
+    index = spot.to_i - 1
+    cells[index] != " "
+  end
+
+  def valid_move?(spot)
+    ok = ["1","2","3","4","5","6","7","8","9"]
+    if ok.include?(spot) && (taken?(spot) == false)
+      return true
+    else
+      false
+    end
+  end
+
```

Comment:
I'd re-write as:

```ruby
def valid_move?(spot)
  (1..9).to_a.include?(spot.to_i) && !taken?(spot)
end
```

Or

```ruby
def valid_move?(spot)
  spot.to_i.between?(1,9) && !taken?(spot)
end
```

Try to use the fact that a method with no code executed will return nil by default to not have to explicitly return false. 

And then if you can avoid a local variable, that's better. We want to maintain as little state as possible.

And then, use booleans that implicitly return true/false and avoid if statements. This style makes the method more declarative, it is the definition of the logic, not the implementation.